submariner/multiple-active-gateways.md

```diff
@@ -0,0 +1,100 @@
+* Each gateway node will establish a tunnel to each peer gateway node.
+* The gateway nodes will load-balance flows between the active peer gateway nodes.
+* Load balancing will be performed using Linux resilient nexthop groups.
+
+### Multiple Active Gateway Packet Flow
+
+An example packet flow is shown in the following diagram and described below.
+
+![MultiGateway](./images/multi-gateway-flow.png)
+
+1. Node 1-1
+    1. Intercept packet destined for global service
+    2. Load balances flow to gateway 1-1
+    3. Hash ensures remaining packets in the connection also go to gateway 1-1
+2. Gateway 1-1
+    1. Determines packet is destined for Cluster 2 based on dest Globalnet IP
+    2. Load balances flow to Gateway 2-2
+    3. Perform SNAT from local source/port to one of the Globalnet address/ports for gateway 1-1
+    4. Hash ensures that remaining packets in the flow also go to Gateway 2-2
+3. Gateway 2-2
+    1. The packet is handed off to the local service load balancer, which performs necessary NAT
+operations and forwards the packet to chosen endpoint on Node 2-1.
+4. Node 2-1
+    1. Send response back to Gateway 2-2 (through service load balancer)
+    2. Note: it should be okay if the return path uses a different gateway, but I think the cluster
+load balancer will return it to the same node in order to maintain the NAT state.
+5. Gateway 2-2
+    1. Forward back to Gateway 1-1 over the appropriate tunnel
+    2. Note: It is important that Gateway 2-2 forwards the packet directly to Gateway 1-1 and not
+via the nexthop load balancer which may send it to Gateway 1-2. This can be accomplished by using host
+routes for the gateway IP addresses that are higher priority than the routes for the cluster Globalnet
+CIDRs which would use the nexthop load balancing.
+6. Gateway 1-1
+    1. Perform DNAT from the global address back to the local address
+    2. Send to the pod on Node 1-1 via local CNI.
+
+## Notes and Assumptions
+
+* This design assumes that Globalnet is being used. It may be possible to make it work without Globalnet if
+necessary, but step 5 relies on the fact that the dest IP will be the address for the correct gateway.
+Without Globalnet, we’d still need to do the same type of SNAT operation at step 2 using local addresses
+which isn’t currently done by Submariner.
+
+* The initial POC will use standard external IP addresses for gateways, but load balancer mode is
+a high priority for the final implementation.
+
+* The initial design and POC is focused on non-ovn-kubernetes CNIs, but ovn-kubernetes must also be supported.
+
+* There will be a config option to enable multiple active gateways, and when enabled, all nodes labeled with
+“submariner.io/gateway=true” will become active. It is for future study whether we should support an option
+where we support more than one active gateway as well as additional backup inactive gateways.
+
+## Plan
```

Comment:
I don't see any alternatives, nor do I see any detailed design (except a theoretical example packet flow)

Reply:
The details have been included in the updated proposal.